rhp/v4/rhp.go

```diff
@@ -892,8 +892,49 @@ func RenewContract(fc types.V2FileContract, prices HostPrices, rp RPCRenewContra
 	}
 }
 
-// RefreshContract creates a contract renewal for the refresh RPC.
-func RefreshContract(fc types.V2FileContract, prices HostPrices, rp RPCRefreshContractParams) (types.V2FileContractRenewal, Usage) {
+// RefreshContractPartialRollover creates a contract renewal for the refresh RPC v1.
+func RefreshContractPartialRollover(fc types.V2FileContract, prices HostPrices, rp RPCRefreshContractParams) (types.V2FileContractRenewal, Usage) {
+	var renewal types.V2FileContractRenewal
+	renewal.FinalRenterOutput = fc.RenterOutput
```

Comment:
sSetting the final renter and host output here is redundant because they are both overwritten. If we do this for a reason though, for clarity perhaps, does it make sense to rewrite:

```go
if fc.RenterOutput.Value.Cmp(rp.Allowance) > 0 {
    renewal.RenterRollover = rp.Allowance
} else {
    renewal.RenterRollover = fc.RenterOutput.Value
}
renewal.FinalRenterOutput.Value = renewal.FinalRenterOutput.Value.Sub(renewal.RenterRollover)
```

in a way that we use `renewal.FinalRenterOutput` instead of `fc.RenterOutput`. I don't think it does but then we should probably get rid of these two assignments at the top of the function.

Reply:
The renter and host output `Value` field is overwritten. The `Address` field is not. We could set both separately, but I think that makes it more likely we introduce a bug.